drivers/flash/soc_flash_nrf.c

```diff
@@ -153,17 +175,28 @@ static int flash_nrf_erase(struct device *dev, off_t addr, size_t size)
 	u32_t n_pages = size / pg_size;
 	int ret;
 
-	/* Erase can only be done per page */
-	if (((addr % pg_size) != 0) || ((size % pg_size) != 0)) {
-		return -EINVAL;
-	}
+#ifndef CONFIG_SOC_FLASH_NRF_UICR
```

Comment:
Same concern as above, but also using conditionals to select part of a statement is fragile.  In some editors it breaks both automatic indentation management and brace-matching.

Reply:
changes all ne #ifdef-ry to possitive wise